### PR TITLE
STORM-2516: Fix timing issues with testPrepareLateTupleStreamWithoutB…

### PR DESCRIPTION
**Pull request by srdo**
…uilder

The test was sleeping a bit and hoping that WaterMarkEventGenerator.run() had been run by the executor in that class before the test woke up. Sometimes that wasn't happening. Instead we're calling the method manually now, like the rest of the tests involving that class.

I also fixed the checkstyle violations in WaterMarkEventGenerator.

As far as I can tell, the lastWatermarkTs variable in WaterMarkEventGenerator should be volatile or an AtomicLong, since it is being written to in run() with the expectation that it can be seen by the thread calling track().

### CODE REVIEW

File: storm-client/src/jvm/org/apache/storm/windowing/WaterMarkEventGenerator.java [hunk 1]
```diff
@@ -45,17 +46,25 @@
     private final Map<GlobalStreamId, Long> streamToTs;
     private final ScheduledExecutorService executorService;
     private final int interval;
+    private final AtomicLong lastWaterMarkTs;
```
Comment (arunmahadevan) [inline storm-client/src/jvm/org/apache/storm/windowing/WaterMarkEventGenerator.java]: this could be a volatile instead of AtomicLong

Comment (srdo) [inline storm-client/src/jvm/org/apache/storm/windowing/WaterMarkEventGenerator.java]: Yes. Why do you prefer volatile? (I'll be happy to change it, I'm just wondering if there's a drawback to using AtomicX)

Comment (arunmahadevan) [inline storm-client/src/jvm/org/apache/storm/windowing/WaterMarkEventGenerator.java]: It doesn't make much difference except the extra overhead of the AtomicLong object and we don't require the additional atomic operations from AtomicLong.

Comment (srdo) [inline storm-client/src/jvm/org/apache/storm/windowing/WaterMarkEventGenerator.java]: Replaced with volatile

**Comment by arunmahadevan** [comment]
+1

**Pull request merged**